### PR TITLE
Revert "Revert "Creates new colocation for colocate_with:='none' too""

### PR DESCRIPTION
**Pull request by aykut-bozkurt**
DESCRIPTION: Creates new colocation entries when colocate_with => 'none'.

This reverts commit d171a736ab3782f8f4f176e345a145ba520e5f47.

We currently create new colocation entry for a distributed table with the option colocate_with => 'none'. We also drop colocation entries if there is no table left to use it.

### CODE REVIEW

File: src/test/regress/expected/isolation_ref_update_delete_upsert_vs_all_on_mx.out [hunk 1]
```diff
@@ -227,22 +227,6 @@ run_commands_on_session_level_connection_to_node
 
 (1 row)
 
-step s2-start-session-level-connection:
-        SELECT start_session_level_connection_to_node('localhost', 57638);
```
Comment (marcocitus) [inline src/test/regress/expected/isolation_ref_update_delete_upsert_vs_all_on_mx.out:231]: is it ok that we're removing this?

Comment (aykut-bozkurt) [reply src/test/regress/expected/isolation_ref_update_delete_upsert_vs_all_on_mx.out:231]: I did not check but the removal of those lines comes from the reversal commit itself which is approved before.

File: src/backend/distributed/metadata/metadata_cache.c [hunk 2]
```diff
@@ -615,6 +615,45 @@ PartitionColumnViaCatalog(Oid relationId)
...
+		table_close(pgDistPartition, NoLock);
+		return INVALID_COLOCATION_ID;
+	}
+
+	Datum colocationIdDatum = datumArray[Anum_pg_dist_partition_colocationid - 1];
+	uint32 colocationId = DatumGetUInt32(colocationIdDatum);
+
+	heap_freetuple(partitionTuple);
+	table_close(pgDistPartition, NoLock);
+
```
[the remark this answers is not in the document]
Comment (aykut-bozkurt) [reply src/backend/distributed/metadata/metadata_cache.c:652]: We actually do not change any tuple there, but only look up a tuple. I think it is ok to not have it. What do you think?

[the remark this answers is not in the document]
Comment (marcocitus) [reply src/backend/distributed/metadata/metadata_cache.c:652]: ah, you're right, nvm

**Pull request merged**